src/js/editor/editor.js

```diff
@@ -344,7 +344,7 @@ class Editor {
       // Above logic might delete redundant range, so callback must run after it.
       let defaultPrevented = false;
       const event = { preventDefault() { defaultPrevented = true; } };
-      this.runCallbacks(CALLBACK_QUEUES.WILL_HANDLE_NEWLINE, [event]);
+      this.runCallbacks(CALLBACK_QUEUES.WILL_HANDLE_NEWLINE, [event, this]);
```

Comment:
I still don't think we need to pass the editor again. It will be in scope in the user's code already.